nifi-nar-bundles/nifi-dropbox-bundle/nifi-dropbox-processors/src/test/java/org/apache/nifi/processors/dropbox/ListDropboxIT.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.nifi.processors.dropbox;
+
+import static java.util.stream.Collectors.toList;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.Spliterator;
+import java.util.Spliterators;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+import org.apache.nifi.json.JsonRecordSetWriter;
+import org.apache.nifi.reporting.InitializationException;
+import org.apache.nifi.serialization.RecordSetWriterFactory;
+import org.apache.nifi.util.MockFlowFile;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class ListDropboxIT extends AbstractDropboxIT<ListDropbox>{
+
+    private static final String YOUNG_FILE_NAME = "just_created" ;
+
+    @BeforeEach
+    public void init() throws Exception {
+        super.init();
+        testRunner.setProperty(ListDropbox.FOLDER_NAME, MAIN_FOLDER);
+    }
+
+    @Override
+    protected ListDropbox createTestSubject() {
+        return new ListDropbox();
+    }
+
+    @Test
+    void testEmbeddedDirectoriesAreListed() throws Exception {
+        // GIVEN
+        createFile("test_file1", "test_file_content1", MAIN_FOLDER);
+        createFile("test_file2", "test_file_content2", MAIN_FOLDER);
+        createFile("test_file11", "test_file_content11", MAIN_FOLDER + "/testFolder1");
+        createFile("test_file112", "test_file_content112", MAIN_FOLDER + "/testFolder2");
+
+        createFile("test_file_not_in_main_folder", "test_file_content31", "/notMainFolder");
+
+        Set<String> expectedFileNames = new HashSet<>(Arrays.asList("test_file1", "test_file2", "test_file11",
+                "test_file112"));
+
+        // The creation of the files are not (completely) synchronized.
+        Thread.sleep(2000);
+
+        // WHEN
+        testRunner.run();
+
+        // THEN
```

Comment:
The `GIVEN / WHEN / THEN` comments do not provide much value and should be removed.